frontend/src/contexts/TokenContext.tsx

```diff
@@ -0,0 +1,20 @@
+/* eslint-disable react-refresh/only-export-components */
+import { tokenPreviews } from '@/tokens'
+import { TokenPreview } from '@/types'
+import { createContext, ReactNode, useContext, useState } from 'react'
+
+interface TokenContextProps {
+  selectedToken: TokenPreview
+  setSelectedToken: React.Dispatch<React.SetStateAction<TokenPreview>>
+}
+
+const TokenContext = createContext<TokenContextProps | null>(null)
+
+export const useTokenContext = () => {
+  return useContext(TokenContext)
```

Comment:
We could throw an error (or assert using `common-universal`) if the context doesn't exist, later one we could remove `!` from `useTokenContext`. It's okay as it's developer mistake and we should expect the component to be wrapped with the provider